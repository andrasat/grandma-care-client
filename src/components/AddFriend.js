import React from 'react'
import { ToastAndroid, View, ToastAndroid } from 'react-native'
import { Container, Content, Header, Left, Body, Right, Input, Item, Icon, Text, Button, Label, Form } from 'native-base'

import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import { addFriend, fetchOneUser, addFriendDone } from '../actions'

class AddFriend extends React.Component {

  state = {
    friendID: ''
  }

  handleFriendIDInput = text => {
    this.setState({
      friendID: text
    })
  }

  componentWillUnmount() {
    const { token, userID, fetchOneUser, addFriendState } = this.props
    fetchOneUser(token,userID)

    if(addFriendState.hasOwnProperty('added')) {
      if(addFriendState.added) {
        ToastAndroid.showWithGravity('Friend added !', ToastAndroid.SHORT, ToastAndroid.CENTER)

      } else {
        ToastAndroid.showWithGravity('Error', ToastAndroid.SHORT, ToastAndroid.CENTER)
      }
    }
  }

  render() {
    const { _setModalAddFriendVisible, token, userID } = this.props
    const { friendID } = this.state
    return (
      <Container>
        <Header>
          <Left>
            <Button light transparent iconLeft
              onPress={()=> _setModalAddFriendVisible(false)}>
              <Icon name="arrow-back" color="#FFF"/>
              <Text>Back</Text>
            </Button>
          </Left>
        </Header>
        <Content>
          <View style={{marginTop:'45%',alignItems:'center'}}>
            <Form style={{width:'90%'}}>
              <Item floatingLabel>
                <Label>Friend ID</Label>
                <Input
                  autoCapitalize="none"
                  autoCorrect={false}
                  value={friendID}
                  onChange={(e) => this.handleFriendIDInput(e.nativeEvent.text)}
                  />
              </Item>
              <Item last style={{marginTop:20, borderColor:'transparent'}}>
                <Button bordered
                  onPress={() => {
                    addFriend(token,userID,friendID)
                    setTimeout(()=> {
                      _setModalAddFriendVisible(false)
                    })
                  }}>
                  <Icon name="add" android="md-add" color="#292988"/>
                </Button>
              </Item>
            </Form>
          </View>
        </Content>
      </Container>
    )
  }
}

AddFriend.propTypes = {
  addFriend: PropTypes.func.isRequired,
  addFriendDone: PropTypes.func.isRequired,
  fetchOneUser: PropTypes.func.isRequired,
  addFriendState: PropTypes.object.isRequired,
  token: PropTypes.string.isRequired,
  userID: PropTypes.string.isRequired
}

const mapStateToProps = state => ({
  addFriendState: state.addFriendState
})

const mapDispatchToProps = dispatch => ({
  addFriend: (token, userID, friendID) => dispatch(addFriend(token,userID,friendID)),
  addFriendDone: () => dispatch(addFriendDone())
  fetchOneUser: (token,userID) => dispatch(fetchOneUser(token,userID))
})

export default connect(mapStateToProps, mapDispatchToProps)(AddFriend)