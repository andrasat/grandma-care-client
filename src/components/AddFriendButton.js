import React from 'react'
import { Button, Text, Icon } from 'native-base'

import PropTypes from 'prop-types'

class AddFriendButton extends React.Component {

  render() {
    return (
      <Button light transparent vertical
        onPress={() => this.props._setModalAddFriendVisible(true)}>
        <Icon name="person-add" android="md-person-add" color="#FFF"/>
        <Text>Add Care</Text>
      </Button>
    )
  }
}

AddFriendButton.propTypes = {
  _setModalAddFriendVisible: PropTypes.func.isRequired
}

export default AddFriendButton