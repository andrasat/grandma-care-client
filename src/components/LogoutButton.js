import React from 'react'
import { AsyncStorage, Alert, ToastAndroid } from 'react-native'
import { Button, Text } from 'native-base'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import { logout } from '../actions'

class LogoutButton extends React.Component {

  _logout = async () => {
    try {
      await AsyncStorage.removeItem('token')
      ToastAndroid.showWithGravity(
        'Logging out',
        ToastAndroid.LONG,
        ToastAndroid.CENTER
      )
    } catch (err) {
      Alert.alert(
        'Logout Error',
        'Press Logout Button again'
      )
    }
  }

  render() {
    const { navigation } = this.props
    return (
      <Button light transparent
        onPress={()=> {
          this._logout()
          setTimeout(()=> {
            navigation.goBack()
          }, 750)
        }}>
        <Text>Logout</Text>
      </Button>
    )
  }

}

LogoutButton.propTypes = {
  navigation: PropTypes.object.isRequired
}

export default LogoutButton