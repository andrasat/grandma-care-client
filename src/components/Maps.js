import React from 'react'
import { Alert, PermissionsAndroid } from 'react-native'
import MapView from 'react-native-maps'
import BackgroundTimer from 'react-native-background-timer'
import PropTypes from 'prop-types'

import { styles } from '../styles'

class Maps extends React.Component {
  constructor(props){
    super(props)
    this.state = {
      idTimer: null,
      latitude: 0,
      longitude: 0,
    }
    this.props.updateLocation=this.props.updateLocation.bind(this);
  }

  const intervalId = (lat,long,userID,token) => {
    return BackgroundTimer.setInterval(() => {
      navigator.geolocation.getCurrentPosition(
        (position) => {
          this.setState({
            latitude: position.coords.latitude,
            longitude: position.coords.longitude
          })
          this.props.updateLocation(lat,long,userID,token)
        },
        (error) => Alert.alert('Turn on GPS',JSON.stringify(error)),
        {timeout: 5000, maximumAge: 2000}
      );
    }, 1000);
  }



  watchID: ?number = null

  componentDidMount() {
    intervalId(this.state.latitude, this.state.longitude,this.props.userID,this.props.token)
  }

  render() {
    const { user } = this.props
    const { latitude, longitude } = this.state
    return (
      <MapView
        style={styles.map}
        region={{
          latitude: latitude,
          longitude: longitude,
          latitudeDelta: 0.0922,
          longitudeDelta: 0.0421
        }}>
        <MapView.Marker
          coordinate={{ latitude, longitude }}
          title={user.username}
          description={"test"}/>
      </MapView>
    )
  }
}

Maps.propTypes = {
  user: PropTypes.object.isRequired
}

const mapDispatchToProps = dispatch => ({
  updateLocation: (lat,long,userID,token) => dispatch(updateLocation(latitude,longitude,userID,token))
})

export default connect(null,mapDispatchToProps)(Maps);
