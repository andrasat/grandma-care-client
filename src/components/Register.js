import React from 'react'
import { View, Text, Alert } from 'react-native'
import { Container, Content, Form, Item, Button, Label, Input } from 'native-base'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import { register, registerDone } from '../actions'
import { styles } from '../styles'

class Register extends React.Component {

  state = {
    username: '',
    password: '',
    phone: ''
  }

  handleUsernameInput = text => {
    this.setState({
      username: text
    })
  }
  handlePasswordInput = text => {
    this.setState({
      password: text
    })
  }
  handlePhoneInput = text => {
    this.setState({
      phone: text
    })
  }

  componentWillUnmount() {
    if(this.props.registerState.hasOwnProperty('register')){
      if(this.props.registerState.register){
        Alert.alert('Register Success')
        this.props.registerDone()
      } else {
        Alert.alert('Register Failed')
      }
    }
  }

  render() {
    const { username, password, phone } = this.state
    const { register, _setModalVisible } = this.props
    return (
      <Container>
        <Content>
          <View style={{marginTop:'45%', alignItems:'center'}}>
            <View>
              <Text style={{fontSize:25}}>Register Here</Text>
            </View>
            <Form style={{width:'85%'}}>
              <Item floatingLabel>
                <Label>Username</Label>
                <Input
                  autoCapitalize="none"
                  autoCorrect={false}
                  value={username}
                  onChange={e => this.handleUsernameInput(e.nativeEvent.text)}
                />
              </Item>
              <Item floatingLabel>
                <Label>Password</Label>
                <Input
                  autoCapitalize="none"
                  autoCorrect={false}
                  secureTextEntry={true}
                  value={password}
                  onChange={e => this.handlePasswordInput(e.nativeEvent.text)}
                />
              </Item>
              <Item floatingLabel>
                <Label>Phone</Label>
                <Input
                  autoCapitalize="none"
                  autoCorrect={false}
                  value={phone}
                  onChange={e => this.handlePhoneInput(e.nativeEvent.text)}
                />
              </Item>
              <Item>
                <Button full style={{width:'100%', marginTop:15}}
                  onPress={()=> {
                    register({username,password,phone})
                    setTimeout(()=> {
                      _setModalVisible(false)
                    }, 500)
                  }}>
                  <Text>Register</Text>
                </Button>
              </Item>
            </Form>
          </View>
        </Content>
      </Container>
    )
  }

}

Register.propTypes = {
  register: PropTypes.func.isRequired,
  registerDone: PropTypes.func.isRequired,
  _setModalVisible: PropTypes.func.isRequired,
  registerState: PropTypes.object.isRequired
}

const mapStateToProps = state => ({
  registerState: state.registerState
})

const mapDispatchToProps = dispatch => ({
  register: user => dispatch(register(user)),
  registerDone: () => dispatch(registerDone())
})

export default connect(mapStateToProps, mapDispatchToProps)(Register)