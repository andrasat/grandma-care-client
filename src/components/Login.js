import React from 'react'
import { View, Modal, Text, AsyncStorage, Alert } from 'react-native'
import { Container, Content, Form, Item, Button, Input, Label} from 'native-base'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import { login } from '../actions'
import { styles } from '../styles'
import Register from './Register'

class Login extends React.Component {

  static navigationOptions = {
    title: 'Login'
  }

  state = {
    username: '',
    password: '',
    modalVisible: false
  }

  componentWillMount() {
    let self = this

    AsyncStorage.getItem('token', (err,result) => {
      if(result !== null){
        self.props.navigation.navigate('Main')
      }
    })
  }

  handleUsernameInput = text => {
    this.setState({
      username: text
    })
  }
  handlePasswordInput = text => {
    this.setState({
      password: text
    })
  }

  _setModalVisible = val => {
    this.setState({
      modalVisible: val
    })
  }

  _handleLogin= (input) => {
    const { login, navigation } = this.props

    login(input)
    navigation.navigate('SplashScreen')
  }

  render() {
    const { username, password, modalVisible } = this.state
    const { login, navigation } = this.props
    return (
      <Container>
        <Content>
          <View style={{flex:1,marginTop:'50%',alignItems:'center'}}>
            <View>
              <Text style={{fontSize:25}}>Login Here</Text>
            </View>
            <Form style={{width:'85%'}}>
              <Item floatingLabel>
                <Label>Username</Label>
                <Input
                  autoCapitalize="none"
                  autoCorrect={false}
                  value={username}
                  onChange={e => this.handleUsernameInput(e.nativeEvent.text)}
                />
              </Item>
              <Item floatingLabel>
                <Label>Password</Label>
                <Input
                  autoCapitalize="none"
                  autoCorrect={false}
                  secureTextEntry={true}
                  value={password}
                  onChange={e => this.handlePasswordInput(e.nativeEvent.text)}
                />
              </Item>
              <Item last style={{marginTop:20, borderColor:'transparent'}}>
                <Button block
                  onPress={(e) => {
                    e.preventDefault()
                    this._handleLogin({username,password})
                  }}>
                  <Text>Login</Text>
                </Button>
                <Button transparent primary
                  onPress={() => this._setModalVisible(true)}>
                  <Text>or Register here</Text>
                </Button>
              </Item>
            </Form>
          </View>
        </Content>
        <Modal
          animationType={"slide"}
          transparent={false}
          visible={modalVisible}
          onRequestClose={() => this._setModalVisible(false)}>
          <Register _setModalVisible={this._setModalVisible} />
        </Modal>
      </Container>
    )
  }

}

Login.propTypes = {
  login: PropTypes.func.isRequired,
  registerState: PropTypes.object.isRequired
}

const mapStateToProps = state => ({
  registerState: state.registerState
})

const mapDispatchToProps = dispatch => ({
  login: user => dispatch(login(user))
})

export default connect(mapStateToProps, mapDispatchToProps)(Login)