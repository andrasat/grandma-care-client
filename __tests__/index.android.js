import 'react-native';
import React from 'react';
import client from '../index.android.js';
import AddFriend from '../src/components/AddFriend.js';
import AddFriendButton from '../src/components/AddFriendButton.js';
import Login from '../src/components/Login.js';
import LogoutButton from '../src/components/LogoutButton.js';
import Main from '../src/components/Main.js';
import Maps from '../src/components/Maps.js';
import Register from '../src/components/Register.js';
import SettingPage from '../src/components/SettingPage.js';
import SplashScreen from '../src/components/SplashScreen.js';
import UserList from '../src/components/UserList.js';
import UserListButton from '../src/components/UserListButton.js';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders with snapshot', () => {
  const tree = renderer.create(<client />);
  expect(tree).toMatchSnapshot();
});

it('renders AddFriend snapshot', () => {
  const compAddFriend = renderer.create(<AddFriend/>);
  expect(compAddFriend).toMatchSnapshot();
});

it('renders AddFriendButton snapshot', () => {
  const compAddFriendButton = renderer.create(<AddFriendButton/>);
  expect(compAddFriendButton).toMatchSnapshot();
});

it('renders Login snapshot', () => {
  const compLogin = renderer.create(<Login/>);
  expect(compLogin).toMatchSnapshot();
});

it('renders LogoutButton snapshot', () => {
  const compLogoutButton = renderer.create(<LogoutButton/>);
  expect(compLogoutButton).toMatchSnapshot();
});

it('renders Main snapshot', () => {
  const compMain = renderer.create(<Main/>);
  expect(compMain).toMatchSnapshot();
});

it('renders Maps snapshot', () => {
  const compMaps = renderer.create(<Maps/>);
  expect(compMaps).toMatchSnapshot();
});

it('renders Register snapshot', () => {
  const compRegister = renderer.create(<Register/>);
  expect(compRegister).toMatchSnapshot();
});

it('renders SettingPage snapshot', () => {
  const compSettingPage = renderer.create(<SettingPage/>);
  expect(compSettingPage).toMatchSnapshot();
});

it('renders SplashScreen snapshot', () => {
  const compSplashScreen = renderer.create(<SplashScreen/>);
  expect(compSplashScreen).toMatchSnapshot();
});

it('renders UserList snapshot', () => {
  const compUserList = renderer.create(<UserList/>);
  expect(compUserList).toMatchSnapshot();
});

it('renders UserListButton snapshot', () => {
  const compUserListButton = renderer.create(<UserListButton/>);
  expect(compUserListButton).toMatchSnapshot();
});
