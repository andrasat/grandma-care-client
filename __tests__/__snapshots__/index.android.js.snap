// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AddFriend snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
  virtual={undefined}
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderBottomColor": "#a7a6ab",
          "borderBottomWidth": 0.5,
          "elevation": 3,
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "center",
          "left": 0,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 15,
          "right": 0,
          "top": 0,
        }
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
          }
        }
        virtual={undefined}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": null,
              "backgroundColor": null,
              "borderBottomWidth": null,
              "borderColor": null,
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "marginLeft": -3,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 6,
              "paddingRight": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#FFF",
                  "fontSize": 12,
                },
                Object {
                  "color": "#f4f4f4",
                  "fontSize": 33,
                  "marginLeft": 0,
                  "marginRight": 10,
                  "marginTop": 2,
                  "paddingTop": 2,
                },
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
            virtual={undefined}
          >
            Ôèè
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#f4f4f4",
                "fontFamily": "System",
                "fontSize": 17,
                "lineHeight": 19,
                "marginLeft": 0,
                "marginRight": 0,
              }
            }
            virtual={undefined}
          >
            Back
          </Text>
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardDismissMode="interactive"
    onScroll={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={0}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    virtual={undefined}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginTop": "45%",
          }
        }
      >
        <View
          style={
            Object {
              "width": "90%",
            }
          }
          virtual={undefined}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 1,
                "borderColor": "#D9D5DC",
                "borderLeftWidth": 0,
                "borderRightWidth": 0,
                "borderTopWidth": 0,
                "flexDirection": "row",
                "marginLeft": 15,
                "marginTop": 15,
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "left": 0,
                  "opacity": 1,
                  "paddingBottom": undefined,
                  "paddingTop": undefined,
                  "position": "absolute",
                  "right": 0,
                  "top": 18,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#575757",
                    "fontSize": 17,
                    "left": 0,
                    "paddingRight": 5,
                    "top": 8,
                  }
                }
                virtual={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontSize": 17,
                    }
                  }
                  virtual={undefined}
                >
                  Friend ID
                </Text>
              </Text>
            </View>
            <TextInput
              autoCapitalize="none"
              autoCorrect={false}
              editable={true}
              onBlur={[Function]}
              onChange={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholderTextColor="#575757"
              style={
                Object {
                  "color": "#000",
                  "flex": 1,
                  "fontSize": 17,
                  "height": 60,
                  "lineHeight": 24,
                  "paddingLeft": null,
                  "paddingRight": 5,
                  "top": 10,
                }
              }
              underlineColorAndroid="rgba(0,0,0,0)"
              value=""
              virtual={undefined}
            />
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 1,
                "borderColor": "transparent",
                "borderLeftWidth": 0,
                "borderRightWidth": 0,
                "borderTopWidth": 0,
                "flexDirection": "row",
                "marginLeft": 0,
                "marginTop": 20,
                "opacity": 1,
                "paddingLeft": 15,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              isTVSelectable={true}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "flex-start",
                  "backgroundColor": "transparent",
                  "borderBottomWidth": 1,
                  "borderColor": "#007aff",
                  "borderLeftWidth": 1,
                  "borderRadius": 5,
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                  "borderWidth": 1,
                  "elevation": null,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "space-between",
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 6,
                  "shadowColor": null,
                  "shadowOffset": null,
                  "shadowOpacity": null,
                  "shadowRadius": null,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                android="md-add"
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#292988",
                      "fontSize": 12,
                    },
                    Object {
                      "color": "#007aff",
                      "fontSize": 24,
                      "marginLeft": 5,
                      "marginRight": 5,
                      "paddingTop": 2,
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
                virtual={undefined}
              >
                ÔÑÇ
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    supportedOrientations={null}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-start",
          "margin": 20,
        }
      }
    >
      <View
        danger={false}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0,0,0,0.8)",
            "borderRadius": 5,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "minHeight": 50,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          }
        }
        success={false}
        virtual={undefined}
        warning={false}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "flex": 1,
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          virtual={undefined}
        />
      </View>
    </View>
  </Modal>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.4)",
          "flex": 1,
          "justifyContent": "flex-end",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "elevation": 4,
            "height": NaN,
            "opacity": 1,
            "padding": 15,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#757575",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          virtual={undefined}
        />
        <RCTScrollView
          dataArray={Array []}
          dataSource={
            ListViewDataSource {
              "items": 0,
            }
          }
          enableEmptySections={true}
          renderRow={[Function]}
          renderScrollComponent={[Function]}
          style={
            Object {
              "marginHorizontal": -15,
              "marginTop": 15,
            }
          }
          virtual={undefined}
        >
          <View />
        </RCTScrollView>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`renders AddFriendButton snapshot 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": null,
      "borderBottomWidth": null,
      "borderColor": "#007aff",
      "borderLeftWidth": null,
      "borderRadius": 5,
      "borderRightWidth": null,
      "borderTopWidth": null,
      "elevation": 0,
      "flexDirection": "column",
      "height": null,
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingBottom": 6,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 6,
      "shadowColor": null,
      "shadowOffset": null,
      "shadowOpacity": null,
      "shadowRadius": null,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    android="md-person-add"
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#FFF",
          "fontSize": 12,
        },
        Object {
          "color": "#f4f4f4",
          "fontSize": 24,
          "marginLeft": 5,
          "marginRight": 5,
          "paddingTop": 2,
        },
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
    virtual={undefined}
  >
    Ôá†
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#f4f4f4",
        "fontFamily": "System",
        "fontSize": 16.5,
        "lineHeight": 19,
        "marginLeft": 0,
        "marginRight": 0,
      }
    }
    virtual={undefined}
  >
    Add Friend
  </Text>
</View>
`;

exports[`renders LogoutButton snapshot 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": null,
      "borderBottomWidth": null,
      "borderColor": "#007aff",
      "borderLeftWidth": null,
      "borderRadius": 5,
      "borderRightWidth": null,
      "borderTopWidth": null,
      "elevation": 0,
      "flexDirection": "column",
      "height": null,
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingBottom": 6,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 6,
      "shadowColor": null,
      "shadowOffset": null,
      "shadowOpacity": null,
      "shadowRadius": null,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#FFF",
          "fontSize": 12,
        },
        Object {
          "color": "#f4f4f4",
          "fontSize": 24,
          "marginLeft": 5,
          "marginRight": 5,
          "paddingTop": 2,
        },
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
    virtual={undefined}
  >
    ÔáÆ
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#f4f4f4",
        "fontFamily": "System",
        "fontSize": 16.5,
        "lineHeight": 19,
        "marginLeft": 0,
        "marginRight": 0,
      }
    }
    virtual={undefined}
  >
    Logout
  </Text>
</View>
`;

exports[`renders UserListButton snapshot 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": null,
      "borderBottomWidth": null,
      "borderColor": "#007aff",
      "borderLeftWidth": null,
      "borderRadius": 5,
      "borderRightWidth": null,
      "borderTopWidth": null,
      "elevation": 0,
      "flexDirection": "column",
      "height": null,
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingBottom": 6,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 6,
      "shadowColor": null,
      "shadowOffset": null,
      "shadowOpacity": null,
      "shadowRadius": null,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#FFF",
          "fontSize": 12,
        },
        Object {
          "color": "#f4f4f4",
          "fontSize": 24,
          "marginLeft": 5,
          "marginRight": 5,
          "paddingTop": 2,
        },
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
    virtual={undefined}
  >
    ÔëΩ
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#f4f4f4",
        "fontFamily": "System",
        "fontSize": 16.5,
        "lineHeight": 19,
        "marginLeft": 0,
        "marginRight": 0,
      }
    }
    virtual={undefined}
  >
    Friend List
  </Text>
</View>
`;

exports[`renders with snapshot 1`] = `<client />`;
